### PR TITLE
[google-sheets] fix tests

### PR DESCRIPTION
**Pull request by mmelko**
add  SS->DB->SS scenario
refactor google sheet utils to connect create only when necessary
attempt to fix servicenow test
//test `@google-sheets`

### CODE REVIEW

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mmelko** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by avano** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

File: ui-tests/src/test/resources/features/integrations/google-sheets.feature [hunk 1]
```diff
@@ -417,3 +428,80 @@ Feature: Google Sheets Connector
...
+    And set integration name "from-db-to-sheets"
+    And click on the "Publish" button
+    And navigate to the "Integrations" page
+    And wait until integration "from-db-to-sheets" gets into "Running" state
+    #it takes some time to copy that big amount of messages
+    And sleep for jenkins delay or "150" seconds
+    And verify that test sheet contains values on range "A1000:E1000"
+      | 308578 | FL | ST  JOHNS COUNTY | 1377 | 1377 |
+    And verify that test sheet contains values on range "A4999:E4999"
+      | 202669 | FL | LEON COUNTY | 0 | 9270000 |
```
Comment (mcada) [inline ui-tests/src/test/resources/features/integrations/google-sheets.feature]: eof new line missing + maybe keep 2 empty lines before new scenario like we do in other files (more or less)

File: ui-tests/src/test/resources/features/integrations/google-sheets.feature [hunk 2]
```diff
@@ -81,7 +82,7 @@ Feature: Google Sheets Connector
     When select the "Google Sheets" connection
     And select "Append values to a sheet" integration action
     And fill in values
-      | Range | A1:E1 |
+      | Range         | A1:E1 |
```
Comment (avano) [inline ui-tests/src/test/resources/features/integrations/google-sheets.feature]: probably unintended change?

File: ui-tests/src/test/resources/features/integrations/google-sheets.feature [hunk 3]
```diff
@@ -417,3 +428,80 @@ Feature: Google Sheets Connector
     And wait until integration "copy" gets into "Running" state
     Then verify that data test sheet contains values on range "A25000:H25000"
       | 241178 | FL | PALM BEACH COUNTY | 0 | 1810826.38 | 0 | 0 | 1810826.38 |
+
+    @Test
```
Comment (avano) [inline ui-tests/src/test/resources/features/integrations/google-sheets.feature]: this should probably not be here

File: ui-tests/src/test/resources/features/integrations/service-now.feature [hunk 4]
```diff
@@ -44,7 +43,9 @@ Feature: Integration - ServiceNow-amq/log
       | Destination Name | incidents |
       | Destination Type | Queue     |
     And click on the "Next" button
+    And sleep for jenkins delay or "10" seconds
     And click on the "Done" button
+    Then  wait for "blank-slate-pf-icon" element class to be visible
```
Comment (avano) [inline ui-tests/src/test/resources/features/integrations/service-now.feature]: two spaces

File: ui-tests/src/test/resources/features/integrations/service-now.feature [hunk 5]
```diff
@@ -73,8 +74,8 @@ Feature: Integration - ServiceNow-amq/log
     And select "Add Record" integration action
     And select "Add new Incident" from "table" dropdown
     And click on the "Next" button
-    Then sleep for jenkins delay or "15" seconds
-
+    And sleep for jenkins delay or "15" seconds
+    Then  wait for "blank-slate-pf-icon" element class to be visible
```
Comment (avano) [inline ui-tests/src/test/resources/features/integrations/service-now.feature]: two spaces

**Comment by avano** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mmelko** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Pull request merged**